provider/plugin/plugin.go

```diff
@@ -0,0 +1,77 @@
+package plugin
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"sigs.k8s.io/external-dns/endpoint"
+	"sigs.k8s.io/external-dns/plan"
+	"sigs.k8s.io/external-dns/provider"
+)
+
+type PluginProvider struct {
+	provider.BaseProvider
+	client       *http.Client
+	remoteServer string
```

Comment:
likely you want to use `*net.URL` and parse the input to fail early if not parseable.